search.js

```diff
@@ -86,60 +86,44 @@ let pubMedApi = {
 
         pmSvc.fetchSummary(environment, options)
             .then(summaryResults => {
+
+                pmids = summaryResults.result.uids;
+                pmcids = [];
+
+                // digging pmcids out from results
+                for (i = 0; i < pmids.length; i++) { 
+                    for(j = 0; j < summaryResults.result[pmids[i]].articleids.length; j++){
+                        if(summaryResults.result[pmids[i]].articleids[j].idtype == 'pmc'){
+                            pmcids[i] = summaryResults.result[pmids[i]].articleids[j].value;
+                        }
+                    }
+                } 
+
+                // leaving only numeric values (stripping off 'PMC' from front)
+                for (i = 0; i < pmcids.length; i++) {
+                    if(pmcids[i]){
+                        pmcids[i] = pmcids[i].replace(/\D/g, '');    
+                    }
+                } 
+
+                // TODO: decide if things should just fall off if we don't return dynamo results for them
                 Promise
-                    .all(demoSvc
-                        .getDemographicDetailsForIds(summaryResults.result.uids))
+                    .all(demoSvc.getDemographicDetailsForIds(pmcids))
                     .then(demoDetails => {
                         demoDetails.forEach(dd => {
-                            console.log(`processing db query results: ${dd}`);
-                            //add item to results object
-                            let item = summaryResults.result[dd.pmid];
-                            item.male_perc = dd.malePercent;
-                            item.female_perc = dd.femalePercent;
-                            results.items.push(item);
+                            //if demoDetails found, add all fields to item and push to results
```

Comment:
I suspect that implementing this approach of dropping things not in dynmo will result it too much complexity when we try to do it correctly. My gut feeling is that including the results that don't have demographic information will be a lot more robust